### DISCUSSION TITLE
git integration: hide when not listing files in a Git repository

### DISCUSSION THREAD

**Discussion by Jackenmen** [lsd-rs/lsd#846] [Ideas]
Right now, enabling git column causes it to show in all directory listings, regardless of whether we're listing files inside a Git repository or not. In comparison, [`exa`](https://github.com/ogham/exa) only shows the Git column when running it inside a directory that is part of a Git repository (i.e. inside `~/git-repo` and `~/git-repo/subdir` but not inside `~`). I think that exa is better in this regard - it minimizes the clutter that isn't relevant when not run in a repository.

One potential edge case is that it still has to work correctly when run with `--tree` option where only part of a tree might be part of a Git repository.

**Comment by Jackenmen** [comment]
I actually realized that exa just doesn't show the Git integration when run with `--tree` so I suppose that's also an option you may want to consider to make this simpler to implement.

**Reply by Jackenmen** [reply]
`lsd` doesn't handle currently show proper status on subdirectories that are git repositories anyway.

**Comment by zwpaper** [comment]
sound reasonable, we can add this feature later, can you please create an issue to track it?
